### ISSUE TITLE
Add integer predicates and use them in documentation

### ISSUE THREAD

**Issue by mrjbq7** [factor/factor#1563]
It might be useful to add these predicates:

```factor
PREDICATE: non-negative-integer < integer 0 >= ;
PREDICATE: negative-integer < integer 0 < ;
PREDICATE: positive-integer < integer 0 > ;
```

And use them in documentation such as ``iota``, ``bit-count``, ``log2``, etc.

**Comment by erg** [comment]
How about words to see if integers fit in the C types? Do we have that already?

``-3 uint?``

**Comment by mrjbq7** [comment]
```-5 uint c-type-interval between?```

**Comment by jonenst** [comment]
How are predicates implemented by the optimizing compiler ? Does it have to test all the predicates at every generic call ?

**Comment by jonenst** [comment]
From the docs 
> Each predicate must be defined as a subclass of some other class. This ensures that predicates inheriting from disjoint classes do not need to be exhaustively tested during method dispatch.

Okay so you get predicate overhead on a generic word only if you define methods on predicate classes for this word ?

**Comment by bjourne** [comment]
Yes. The dispatch engine creates a kind of dispatch tree that first checks if the superclasses match and only then runs the predicate tests if there are predicate class methods defined.
